### PR TITLE
feat: drain node on shutdown

### LINKED ISSUE #4618: Drain the node on shutdown event.
## Feature Request

Better to make drain node (Kubernetes method) on shutdown event. 
Currently Talos just kills(stop) all pods...

For example you have spot instance, and cloud provider trying to stop the node.
Talos will drain node, and all pods gracefully migrate to another instance.

### PR DESCRIPTION
**Pull request by TimJones**
Cordon & drain a node when the Shutdown message is received.
Also adds a '--force' option to the shutdown command in case the control
plane is unresponsive.

Signed-off-by: Tim Jones <timniverse@gmail.com>

# Pull Request

## What? (description)

Resolves #4618 

## Acceptance

Please use the following checklist:

- [x] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4895)

### CODE REVIEW

**Comment by smira** [comment]
@TimJones looks like we need `make docs` to be re-run

and I think we'll need `make api-descriptors` because there's a seemingly breaking change on the shutdown API

**Comment by TimJones** [comment]
@smira Regenerated docs & api descriptors

File: pkg/machinery/client/client.go [hunk 1]
```diff
@@ -532,8 +532,10 @@ func (c *Client) Bootstrap(ctx context.Context, req *machineapi.BootstrapRequest
 }
 
 // Shutdown implements the proto.MachineServiceClient interface.
-func (c *Client) Shutdown(ctx context.Context) (err error) {
-	resp, err := c.MachineClient.Shutdown(ctx, &emptypb.Empty{})
+func (c *Client) Shutdown(ctx context.Context, force bool) (err error) {
```
Comment (smira) [inline pkg/machinery/client/client.go:535]: we might need to add option arguments here to maintain backwards compatibility, I can take care of that

Comment (smira) [reply pkg/machinery/client/client.go]: did the update

**Review by smira — APPROVED** [review]

**Pull request merged**